src/Shared/Contracts/IPackageExistence.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation. Licensed under the MIT License.
+
+namespace Microsoft.CST.OpenSource.Contracts;
+
+using Model.Enums;
+using System.Collections.Generic;
+
+/// <summary>
+/// An interface representing the existence of a package/version.
+/// </summary>
+public interface IPackageExistence
+{
+    /// <summary>
+    /// Gets if the package/version currently exists.
+    /// </summary>
+    bool Exists { get; }
+
+    /// <summary>
+    /// Gets if the package/version had existed at one point.
+    /// </summary>
+    bool HasEverExisted { get; }
+
+    /// <summary>
+    /// Gets if the package/version was removed.
+    /// So if it doesn't exist now, but at one point did exist.
+    /// </summary>
+    bool WasRemoved => !this.Exists && this.HasEverExisted;
+
+    /// <summary>
+    /// Gets the reasons (if any) for why a package/version was removed.
+    /// </summary>
+    IReadOnlySet<PackageRemovalReason>? RemovalReasons { get; }
```

Comment:
This property is only relevant for the case of a package version having been removed. Given that, does it need to be exposed at the interface level - i.e., could users do a [pattern-matching type test](https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/functional/pattern-matching#type-tests) to get access to this field only if/when the package version was removed?